docs/joss/paper.md

```diff
@@ -0,0 +1,211 @@
+to the model constructor then implements this new model component without
+the need to branch off or overwrite existing model components.
+Conceptually similar scientific modelling paradigms have been very successful
+in the Python-based generic partial differential equation solver Dedalus [@Burns2020]
+and the Julia ocean model Oceananigans.jl [@Ramadhan2020].
+
+![Surface temperature simulated with the primitive equation model in SpeedyWeather.jl.
+(Figure will be updated) \label{fig:primitive}](primitive.jpg)
+
+The dynamical core of SpeedyWeather.jl uses established numerics
+[@Bourke1972; @Hoskins1975; @Simmons1978; @Simmons1981],
+widely adopted in numerical weather prediction. It is based on the spherical
+harmonic transform with a leapfrog-based semi-implicit time integration [@Hoskins1975]
+and a Robert-Asselin-Williams filter [@Williams2011; @Amezcua2011].
+The spherical harmonic transform is grid-flexible. Any iso-latitude ring-based
+grid can be used and new grids can be externally defined and passed in
+as an argument. Many grids are already implemented: the conventional
+Gaussian grid, a regular longitude-latitude grid, 
+the octahedral Gaussian grid [@Malardel2016], the octahedral
+Clenshaw-Curtis grid [@Hotta2018], and the HEALPix grid [@Gorski2005].
+Both SpeedyWeather.jl and its spherical harmonic transform SpeedyTransforms are also
+number format-flexible. 32-bit single-precision floating-point numbers
+(Float32) are the default as adopted by other modelling efforts [@Vana2017; @Nakano2018],
+but Float64 and other custom number formats can be used with a single
+code basis [@Klower2022; @Klower2020].
+Julia will compile to the choice of the number format, the grid,
+and and other model components just-in-time. A simple parallelisation
+across vertical layers is supported with Julia's multi threading.
+Output is stored as NetCDF files using
+[NCDatasets.jl](https://github.com/Alexander-Barth/NCDatasets.jl).
+
+# Statement of need
+
+SpeedyWeather.jl is a fresh approach to atmospheric
+models that have been very influential in many areas of scientific 
+and high-performance computing as well as climate change mitigation and adaptation.
+Most weather, ocean and climate models are written in Fortran and have been developed over
+decades. From this tradition follows a specific programming style and
+associated user interface. Running a simulation in Fortran and analysing the
+data in Python makes it virtually impossible to interact with various model
+components interactively. Furthermore, data-driven climate modelling
+[@Rasp2018; @Schneider2023], which replaces existing model components with machine learning
+is difficult due to the lack of established deep learning frameworks in Fortran [@Innes2019].
+Let alone online learning, which trains a neural network-based component together
+with the rest of the model, accounting for interactions between components.
+Gradients, necessary to optimize training, can be computed 
```

Comment:
```suggestion
Gradients, necessary to optimize neural network weights during training, can be computed 
```
Training is the process of optimization rather than the thing which is optimized.